### ISSUE TITLE
git push to repo breaks because phab daemons lack request headers

### ISSUE THREAD

**Issue by zarvox** [liamzdenek/phabricator-sandstorm#12]
steps to reproduce:
- create a Phabricator instance
- create a Diffusion repository
- clone it over HTTP as described in https://github.com/liamzdenek/phabricator-sandstorm/issues/5#issuecomment-246090821
- create a commit
- push that commit back to Phabricator
- try to browse that repository in Diffusion

Then, in the UI, you see details stuck on "_Importing..._"

![screenshot1](https://cloud.githubusercontent.com/assets/307325/18496135/489c2426-79d8-11e6-831e-93a2177d46db.png)

Looking in the grain logs, you find in `/opt/sandstorm/var/sandstorm/grains/RWXEZSpB9KB7MKR5b6tPZB/sandbox/tmp/phd/log/daemons.log` the following:

```
[14-Sep-2016 00:27:46 UTC] PHP Fatal error:  Call to undefined function getallheaders() in /opt/app/phabricator/src/extensions/PhabricatorSandstormAuthProvider.php on line 30
```

It looks like we'll need to figure out a way to not break when running in the CLI context.

**Comment by liamzdenek** [comment]
Ah so I was crashing the Daemons. I can't possibly do authentication without headers, and I can't reason why the Daemons would need that functionality. I think some other code may be calling getAdapter coincidentally, and if I just return PhutilEmptyAuthAdapter (when in CLI mode), it'll work just fine.

**Comment by liamzdenek** [comment]
![img](http://i.imgur.com/ObEfeuZ.png)
Resolved in: 06b0a09cc73f4d1aa44bed8ee27c07147c320bfd

This might break in the future, I'm not 100% confident that i haven't broken some unrelated thing, but everything LOOKS fine.
